src/TuDa.CIMS.Web/Components/ShoppingCart/ShoppingSearch.razor.cs

```diff
@@ -31,14 +32,39 @@ private async Task AssetItemSelectedInternal(AssetItem item)
         await AssetItemSelected.InvokeAsync(item);
     }
 
+    /// <summary>
+    /// To filter for different types of Items
+    /// </summary>
+    private AssetItemType? _selectedAssetItemType;
+
     private async Task<IEnumerable<AssetItem>> Search(string nameOrCas, CancellationToken token)
     {
         if (string.IsNullOrWhiteSpace(nameOrCas))
         {
             return [];
         }
 
-        return await _assetItemApi.GetAllAsync(nameOrCas).Match(value => value, err => []);
+        var allItems = await _assetItemApi
+            .GetAllAsync(nameOrCas)
+            .Match(value => value, err => new List<AssetItem>());
+
+        switch (_selectedAssetItemType)
+        {
+            case AssetItemType.Chemical:
+                return allItems.Where(item => item is Chemical);
+
+            case AssetItemType.Consumable:
+                return allItems.Where(item => item is Consumable);
+
+            case AssetItemType.Solvent:
+                return allItems.Where(item => item is Solvent);
+
+            case AssetItemType.GasCylinder:
+                return allItems.Where(item => item is GasCylinder);
+
+            default:
+                return allItems;
+        }
```

Comment:
⛏️ This could be simplified by a switch expression and with OfType()
```suggestion
        return _selectedAssetItemType switch
        {
            AssetItemType.Chemical => allItems.Where(item => item is Chemical and not Solvent);
            AssetItemType.Consumable => allItems.OfType<Consumable>();
            AssetItemType.Solvent => allItems.OfType<Solvent>;
            AssetItemType.GasCylinder => allItems.OfType<GasCylinder;
            _ =>  allItems;
        }
```